File: 3rd Sem/DSA/factorial.java
public class factorial {
    static int fact(int a) {
        if (a <= 1)
            return 1;
        else
            return (a * fact(a - 1));
    }

    public static void main(String[] args) {
        int a = 4;
        int fact = fact(a);
        System.out.println("The factorial of " + a + " = " + fact);
    }
}

File: 5th sem/AI/lab 6/factorial.py
def factorial(n):
    if n == 0:
        # Base case: factorial of 0 is 1
        return 1
    else:
        # Recursive case: n! = n * (n-1)!
        return n * factorial(n-1)

def main():
    # Get user input
    num = int(input("Enter a number: "))
    
    # Calculate factorial
    result = factorial(num)
    
    # Print result
    print(f"The factorial of {num} is {result}")

if __name__ == "__main__":
    main()